modules/cloud-run-v2/variables.tf

```diff
@@ -166,29 +160,31 @@ variable "region" {
   default     = "europe-west1"
 }
 
-variable "revision_annotations" {
-  description = "Configure revision template annotations."
+variable "revision" {
+  description = "Revision template configurations."
   type = object({
-    autoscaling = optional(object({
-      max_scale = number
-      min_scale = number
-    }))
-    cloudsql_instances  = optional(list(string), [])
-    vpcaccess_connector = optional(string)
-    vpcaccess_egress    = optional(string)
-    network_interfaces = optional(object({
-      subnetwork = optional(string)
-      tags       = optional(list(string))
+    name                       = optional(string)
+    gen2_execution_environment = optional(bool)
+    max_concurrency            = optional(number)
+    max_instance_count         = optional(number)
+    min_instance_count         = optional(number)
+    vpc_access = optional(object({
+      connector = optional(string)
```

Comment:
Thinking, but I don't get a good answer for your asks. 
- Validation wouldn't serve as things are. We could check connector & subnet are not both defined, but it could be vpc_connector_create & subnet are.
- I'm avoiding making more complex structures for little value. If a user doesn't know they can't define both a connector and direct vpc egress they don't understand it.
- Adding vpc_connector_create stuff in revision, totally against. A connector is not part of a revision, it is used by.
- Not sure what you mean by aligning those two structures, and how it could help. Not so familiar with CF module but I see like a reduced implementation of a connector there.

If any, I would see adding a 'connector_create'-like field in the revision, to then apply validation. In the end, the connector is independent of a revision but it doesn't serve any purpose if not used by a revision.